haskell/toolchain.bzl

```diff
@@ -27,7 +27,58 @@ load(
 
 _GHC_BINARIES = ["ghc", "ghc-pkg", "hsc2hs", "haddock", "ghci", "runghc", "hpc"]
 
-def _run_ghc(hs, cc, inputs, outputs, mnemonic, arguments, env, params_file = None, progress_message = None, input_manifests = None, extra_name = ""):
+def _create_abi(hs, inputs, outputs, progress_message = None):
+    hs.actions.run_shell(
+        inputs = inputs,
+        outputs = outputs,
+        command =
+            """sed -n "s/^\\s*ABI hash:\\s*\\(\\S*\\)$/\\1/p" {readable_hi_input} > {abi_output}""".format(
```

Comment:
Please obtain `sed` from the POSIX toolchain. See [here](https://github.com/tweag/rules_haskell/blob/2b3c715d57b5f8a36a7b4d2e2f4729d6397e3a8a/haskell/private/actions/link.bzl#L103) for an example.

Reply:
For some reason I do not understand why this comment is not tagged as "outdated". Does your comment still applies to the current version, where `sed` is called within the bash script `ghc-wrapper.sh`?

Reply:
> For some reason I do not understand why this comment is not tagged as "outdated".

Good question, perhaps because I commented on the individual commit and not the overall diff?

> Does your comment still applies to the current version, where `sed` is called within the bash script `ghc-wrapper.sh`?

Good question. I think it would still be preferable to have this and potentially other tools provided by the POSIX toolchain. IIRC we already have such a mechanism in place with the [`path_for_run_ghc`](https://github.com/tweag/rules_haskell/blob/662f1423754a2dfd72f991d3346bdefe793bc696/haskell/toolchain.bzl#L83). @ylecornec AFAIK this is currenlty only used for the Asterius toolchian. Do you think we could also use it to provide POSIX tools for the regular toolchain?

Reply:
Given that it's now part of the GHC wrapper, this can be done in a separate PR.

Reply:
Tests fail for me (NixOS) because of a missing sed. Created an issue - #1800